src/ntlm/mod.rs

```diff
@@ -398,10 +416,20 @@ impl Sspi for Ntlm {
             self.complete_auth_token(&mut [])?;
         }
 
-        SecurityBuffer::find_buffer_mut(message, SecurityBufferType::Token)?; // check if exists
+        let mut encrypted = if let Ok(buffer) = SecurityBuffer::find_buffer_mut(message, SecurityBufferType::Token) {
+            buffer
+        } else {
+            SecurityBuffer::find_buffer_mut(message, SecurityBufferType::Stream)?
+        }
+        .buffer
+        .clone();
```

Comment:
Nit: as far as I know, it’s more idiomatic to write this:
```suggestion
        let security_buffer = let Ok(buffer) = SecurityBuffer::find_buffer_mut(message, SecurityBufferType::Token) {
            buffer
        } else {
            SecurityBuffer::find_buffer_mut(message, SecurityBufferType::Stream)?
        };
        let encrypted = &security_buffer.buffer;
```
If I understand correctly, it’s also not necessary to `.clone()` the buffer at all.

Reply:
This code was copied from the kerberos code, and the changes proposed do not compile, I will leave it as is since it's only a nit.